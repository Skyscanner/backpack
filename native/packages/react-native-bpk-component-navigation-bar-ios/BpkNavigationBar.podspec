PACKAGE_ROOT = 'native/packages/react-native-bpk-component-navigation-bar-ios'.freeze

Pod::Spec.new do |s|
  s.name = 'BpkNavigationBariOS'
  s.version = '0.0.0' # TODO: This should be parsed from package.json
  s.homepage = 'http://github.com/skyscanner/backpack'
  s.platform = :ios, '9.0'
  s.source = { git: 'git@github.com:Skyscanner/backpack.git' }
  s.source_files = "#{PACKAGE_ROOT}/iOS/*.{h,m}"
  s.preserve_paths = "#{PACKAGE_ROOT}/src/**/*.js"
  s.dependency 'React'
end
