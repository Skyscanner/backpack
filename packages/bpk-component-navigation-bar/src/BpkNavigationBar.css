@keyframes bpk-keyframe-spin{100%{transform:rotate(1turn)}}.bpk-navigation-bar{position:relative;display:flex;min-height:3rem;padding:.75rem;flex-direction:row;justify-content:center;background-color:#0770e3;background-color:var(--bpk-navigation-bar-background-color, #0770e3);box-shadow:0 -1px 0 0 #dddde5 inset}.bpk-navigation-bar__title{color:#fff;color:var(--bpk-navigation-bar-title-color, #fff)}.bpk-navigation-bar__leading-item,.bpk-navigation-bar__trailing-item{position:absolute}.bpk-navigation-bar__leading-item{left:.75rem}html[dir='rtl'] .bpk-navigation-bar__leading-item{right:.75rem;left:auto}.bpk-navigation-bar__trailing-item{right:.75rem}html[dir='rtl'] .bpk-navigation-bar__trailing-item{right:auto;left:.75rem}
