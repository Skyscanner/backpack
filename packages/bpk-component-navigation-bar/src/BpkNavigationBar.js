/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow strict */

import React, { type Node, type Element } from 'react';
import PropTypes from 'prop-types';

import { cssModules } from '../../bpk-react-utils';
import BpkText, { TEXT_STYLES } from '../../bpk-component-text';

import STYLES from './BpkNavigationBar.module.scss';

const getClassNames = cssModules(STYLES);

export type Props = {
  id: string,
  title: Node,
  className: ?string,
  leadingButton: ?Element<any>,
  trailingButton: ?Element<any>,
  sticky: ?boolean,
};

const cloneWithClass = (elem: Element<any>, newStyle: string) => {
  const className = getClassNames(elem.props.className, newStyle);
  return React.cloneElement(elem, { ...elem.props, className });
};

const BpkNavigationBar = (props: Props) => {
  const {
    className,
    id,
    leadingButton,
    sticky,
    title,
    trailingButton,
    ...rest
  } = props;

  // If the title is a component that sets its own id we want the aria-labelledby on the nav to match this so it can find the element
  // Otherwise if its just a string we set the id on the title component.
  const titleId =
    typeof title === 'string' ? `${id}-bpk-navigation-bar-title` : id;

  return (
    // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
    <nav
      aria-labelledby={titleId}
      className={getClassNames(
        'bpk-navigation-bar',
        sticky && 'bpk-navigation-bar__sticky',
        className,
      )}
      {...rest}
    >
      {leadingButton &&
        cloneWithClass(leadingButton, 'bpk-navigation-bar__leading-item')}
      {typeof title === 'string' ? (
        <BpkText
          id={titleId}
          textStyle={TEXT_STYLES.heading5}
          className={getClassNames('bpk-navigation-bar__title')}
        >
          {title}
        </BpkText>
      ) : (
        title
      )}
      {trailingButton &&
        cloneWithClass(trailingButton, 'bpk-navigation-bar__trailing-item')}
    </nav>
  );
};

BpkNavigationBar.propTypes = {
  title: PropTypes.node.isRequired,
  className: PropTypes.string,
  leadingButton: PropTypes.element,
  trailingButton: PropTypes.element,
  sticky: PropTypes.bool,
};

BpkNavigationBar.defaultProps = {
  className: null,
  leadingButton: null,
  trailingButton: null,
  sticky: false,
};

export default BpkNavigationBar;
