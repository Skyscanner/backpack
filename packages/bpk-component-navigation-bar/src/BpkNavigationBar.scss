/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2019 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~bpk-mixins/index';

.bpk-navigation-bar {
  position: relative;
  display: flex;
  min-height: $bpk-spacing-xxl + $bpk-spacing-xs;
  padding: $bpk-spacing-sm;
  flex-direction: row;
  justify-content: center;

  @include bpk-themeable-property(background-color, --bpk-navigation-bar-background-color, $bpk-color-white);
  @include bpk-border-bottom-sm($bpk-color-gray-100);

  &__title {
    @include bpk-themeable-property(color, --bpk-navigation-bar-title-color, $bpk-color-gray-700);
  }

  &__leading-item,
  &__trailing-item {
    position: absolute;
  }

  &__leading-item {
    left: $bpk-spacing-sm;

    @include bpk-rtl {
      right: $bpk-spacing-sm;
      left: auto;
    }
  }

  &__trailing-item {
    right: $bpk-spacing-sm;

    @include bpk-rtl {
      right: auto;
      left: $bpk-spacing-sm;
    }
  }
}
