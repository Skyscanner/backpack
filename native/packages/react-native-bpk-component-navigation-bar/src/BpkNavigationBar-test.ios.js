/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import BpkNavigationBar from './BpkNavigationBar';
import BpkNavigationBarBackButtonIOS from './BpkNavigationBarBackButtonIOS';
import BpkNavigationBarTextButtonIOS from './BpkNavigationBarTextButtonIOS';

describe('ios', () => {
  describe('BpkNavigationBar', () => {
    it('should render correctly', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with custom style', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
            style={{ zIndex: 5 }}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with trailing button', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            leadingButton={
              <BpkNavigationBarTextButtonIOS
                title="Cancel"
                onPress={jest.fn()}
              />
            }
            trailingButton={
              <BpkNavigationBarTextButtonIOS
                title="Done"
                emphasize
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with a subtitle view', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title="Backpack"
            subtitleView={<View testID="subtitle-view" />}
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with a title view', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title={<View testID="title-view" />}
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly when title is null', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title={null}
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render correctly with an icon in the title', () => {
      const tree = renderer
        .create(
          <BpkNavigationBar
            title={{ value: 'Checkout', icon: 'lock', iconPosition: 'leading' }}
            leadingButton={
              <BpkNavigationBarBackButtonIOS
                title="Back"
                showTitle
                onPress={jest.fn()}
              />
            }
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
