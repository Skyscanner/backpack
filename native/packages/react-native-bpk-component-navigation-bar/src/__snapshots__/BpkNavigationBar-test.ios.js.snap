// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ios BpkNavigationBar should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "maxWidth": "60%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
              },
              Object {
                "fontWeight": "600",
              },
              Array [
                Object {
                  "color": "rgb(82, 76, 97)",
                },
                null,
              ],
            ]
          }
        >
          Backpack
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly when title is null 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly with a subtitle view 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
      Object {
        "paddingBottom": 16,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "maxWidth": "60%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
              },
              Object {
                "fontWeight": "600",
              },
              Array [
                Object {
                  "color": "rgb(82, 76, 97)",
                },
                null,
              ],
            ]
          }
        >
          Backpack
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      testID="subtitle-view"
    />
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly with a title view 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            null,
            Object {
              "maxHeight": 28,
            },
          ]
        }
        testID="title-view"
      />
    </View>
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly with an icon in the title 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "maxWidth": "60%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "BpkIcon",
                "fontSize": 24,
                "includeFontPadding": false,
              },
              Object {
                "fontSize": 16,
              },
              Array [
                Object {
                  "marginEnd": 4,
                },
                Object {
                  "color": "rgb(82, 76, 97)",
                },
              ],
            ]
          }
        >
          
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
              },
              Object {
                "fontWeight": "600",
              },
              Array [
                Object {
                  "color": "rgb(82, 76, 97)",
                },
                Object {
                  "lineHeight": 24,
                },
              ],
            ]
          }
        >
          Checkout
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly with custom style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
      Object {
        "zIndex": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Back"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "testUri": "../../../packages/react-native-bpk-component-navigation-bar/src/chevron-left.png",
          }
        }
        style={
          Array [
            Object {
              "marginEnd": 8,
            },
            Object {
              "tintColor": "rgb(82, 76, 97)",
            },
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "maxWidth": "60%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
              },
              Object {
                "fontWeight": "600",
              },
              Array [
                Object {
                  "color": "rgb(82, 76, 97)",
                },
                null,
              ],
            ]
          }
        >
          Backpack
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ios BpkNavigationBar should render correctly with trailing button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(243, 242, 245)",
        "flexDirection": "column",
        "paddingHorizontal": 8,
        "shadowColor": "rgb(230, 228, 235)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "space-between",
          "overflow": "visible",
          "paddingTop": 20,
        },
        false,
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Cancel"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginStart": 8,
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Cancel
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "height": 44,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "maxWidth": "60%",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "400",
              },
              Object {
                "fontWeight": "600",
              },
              Array [
                Object {
                  "color": "rgb(82, 76, 97)",
                },
                null,
              ],
            ]
          }
        >
          Backpack
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Done"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginEnd": 8,
          "opacity": 1,
          "zIndex": 2,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(82, 76, 97)",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Object {
              "fontWeight": "600",
            },
            Array [
              Object {
                "color": "rgb(82, 76, 97)",
              },
            ],
          ]
        }
      >
        Done
      </Text>
    </View>
  </View>
</View>
`;
